### PR TITLE
gaa: Add 3p sign in button

### PR DESCRIPTION
**Pull request by mark-yoon**
Added a new GaaGoogle3pSignInButton class. Some duplicated code when checking for allowable origins -- creating PR as a sanity check before writing tests.

### CODE REVIEW

File: examples/google-signin/google-signin-iframe.html [hunk 1]
```diff
@@ -21,5 +21,13 @@
       // This should also include your AMP cache origin(s), if you're using AMP.
       allowedOrigins: ['http://localhost:8000', 'http://localhost:8080'],
     });
+    GaaGoogle3pSignInButton.show({
```
Comment (ChrisAntaki) [inline examples/google-signin/google-signin-iframe.html:24]: Should this be in its own `html` file?

Comment (mark-yoon) [reply examples/google-signin/google-signin-iframe.html]: Good point -- added new html file and corresponding route for the example!

File: examples/sample-pub/publisher.js [hunk 2]
```diff
@@ -134,8 +134,14 @@ function startFlowAuto() {
...
 
   // Provide a shortcut to metering params.
-  if (location.search === '?metering') {
-    location.search = 'gaa_at=g&gaa_ts=99999999&gaa_n=n0nc3&gaa_sig=51g';
+  const urlParams = new URLSearchParams(location.search);
+  if (urlParams.has('metering')) {
+    let newSearch = 'gaa_at=g&gaa_ts=99999999&gaa_n=n0nc3&gaa_sig=51g';
+    // Include 3p sign in param if specified.
+    if (urlParams.get('use3pSignIn') === 'true') {
+      newSearch += '&use3pSignIn=true';
```
Comment (ChrisAntaki) [inline examples/sample-pub/publisher.js:142]: Nice

**Review by ChrisAntaki — APPROVED** [review]

**Review by ChrisAntaki — CHANGES_REQUESTED** [review]
Lets keep the 100% coverage for the `gaa.js` and `gaa-test.js` files
https://app.codecov.io/gh/subscriptions-project/swg-js/compare/1919/diff

**Pull request merged**